### PR TITLE
wip: moving from styleguidist to molgenis-components

### PR DESCRIPTION
**Pull request by mswertz**

### CODE REVIEW

File: apps/.eslintrc.js [hunk 1]
```diff
@@ -25,4 +26,13 @@ module.exports = {
       },
     },
   ],
+
+  globals: {
+    __VUE_SSR_CONTEXT__: "readonly",
+  },
+
+  rules: {
+    "no-unused-vars": "off",
```
Comment (connoratrug) [inline apps/.eslintrc.js:35]: why turn this off ? usually unused var point to error or at least hard to read (maintain) code

File: apps/.eslintrc.js [hunk 2]
```diff
@@ -25,4 +26,13 @@ module.exports = {
...
+    "no-undef": "warn",
```
Comment (connoratrug) [inline apps/.eslintrc.js:36]: same

File: apps/build.gradle [hunk 3]
```diff
@@ -81,7 +81,7 @@ subprojects {
         }
     }
 
-    task buildComponentsShowcase(type: YarnTask, dependsOn: buildStyleguidist) {
```
Comment (connoratrug) [inline apps/build.gradle:84]: why this change please note why this is needed in the commit msg

File: apps/molgenis-components/lib/main.js [hunk 4]
```diff
@@ -47,12 +53,28 @@ import ObjectDisplay from "../src/components/tables/cellTypes/ObjectDisplay.vue"
 import StringDisplay from "../src/components/tables/cellTypes/StringDisplay.vue";
 import TextDisplay from "../src/components/tables/cellTypes/TextDisplay.vue";
 
-import SignInForm from "../src/components/organisms/SignInForm.vue"
-
+import MolgenisSignIn from "../src/components/layout/MolgenisSignIn.vue";
+import ShowMore from "../src/components/layout/ShowMore.vue";
```
Comment (connoratrug) [inline apps/molgenis-components/lib/main.js:57]: Placing basic components into the same very large folder as the compositions makes it hard to locate and navigate components. In addition leading with the non descriptive ( and inconsistent ) 'Molgenis' prefix makes it harder to scan the components tree

File: apps/molgenis-components/package.json [hunk 5]
```diff
@@ -1,11 +1,12 @@
 {
-  "name": "molgenis-components",
+  "name": "@molgenis/molgenis-components",
   "version": "0.0.0",
+  "private": true,
```
Comment (connoratrug) [inline apps/molgenis-components/package.json:4]: why this change please add to commit comment

File: apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/_cohort.vue [hunk 6]
```diff
@@ -12,155 +12,167 @@
...
+        <grid-block heading="Subpopulations" v-if="subpopulations.length">
+          <table-display
+            :isClickable="true"
+            @row-click="$router.push({ path: $event._path })"
+            :columns="[
+              { name: 'name', label: 'Name' },
+              { name: 'description', label: 'Description' },
+              { name: 'numberOfParticipants', label: 'Number of participants' },
+              { name: 'ageGroups', label: 'Age categories' },
+            ]"
```
Comment (chinook25) [inline apps/nuxt-ssr/pages/_schema/ssr-catalogue/cohorts/_cohort.vue:137]: Would be nice to put a variable here (and some other places in this file) to keep the html a bit more readable.

File: apps/molgenis-components/src/components/tables/TableMolgenis.vue [hunk 7]
```diff
@@ -243,7 +243,7 @@ export default {
 </template>
 
 <script>
-  import Client from "../../../client/client.js";
+  import Client from "../../../src/client/client.js";
```
Comment (connoratrug) [inline apps/molgenis-components/src/components/tables/TableMolgenis.vue:246]: remove src , no need to go up and then back down again

File: apps/pages/src/components/ListPages.vue [hunk 8]
```diff
@@ -23,23 +23,19 @@
...
       if (this.session && this.session.settings) {
         return Object.keys(this.session.settings)
-          .filter(key => key.startsWith("page."))
-          .map(key => key.substring(5));
+          .filter((key) => key.startsWith("page."))
+          .map((key) => key.substring(5));
       }
-    }
-  }
+      return [];
```
Comment (chinook25) [inline apps/pages/src/components/ListPages.vue:37]: explicit else would be nice

File: apps/styleguide/src/forms/InputOntology.vue [hunk 9]
```diff
@@ -198,7 +197,7 @@ export default {
     },
     getSelectedChildNodes(term) {
       let result = [];
-      if (term.selected == 'complete') {
+      if (term.selected == "complete") {
```
Comment (chinook25) [inline apps/styleguide/src/forms/InputOntology.vue:200]: ===

File: apps/styleguide/src/forms/InputOntology.vue [hunk 10]
```diff
@@ -314,14 +313,14 @@ export default {
     },
     emitValue() {
       let selectedTerms = Object.values(this.terms)
-        .filter((term) => term.selected == 'complete')
+        .filter((term) => term.selected == "complete")
```
Comment (chinook25) [inline apps/styleguide/src/forms/InputOntology.vue:316]: ===

File: apps/molgenis-components/src/components/display/ImageDisplay.vue [hunk 11]
```diff
@@ -48,24 +48,32 @@ export default {
...
...CH5aMBAtaqQydjdQLxTGYr6siTkxC0InUBCHhHC9wMnyEY0XzL91NH9lunIejKNhzqm/IYGr4WqcJMQHs10vGpI2+rJSCDobT9KEVA22ze4kMEIvaBv9ceaiToItn9oY7cr5aAjnii8U2agPbd+GcDCaiQsc6HHm7DREB7IWIQwbWtWoLYMgS8BbzOIxCDHOPxOo1gIR/KHHlddRnE45sMAe1LsS5sDgH1BTbYIPpGK41gJF2ZPTyrqYf6QXchLeoX9JLPxYQ0j6A1YaNMwzFTjaDlifiz1JFnhxGMesIW8e9DbRHPMO325xKII6JEmx7SCabiX13C8nEIwURuFSzsVYhfon/IJxCxTPIiMEkCVicXrETZsw8iEMOXwp6doGUdhyICHjrAz8y36AQL7lXCkQ8uA2RJ0OKDhEICFjOQZNpUnYC58hN7NvuPQwhWw+E4PULKIWD9saVXRMDjyKbD2RkIeFFRWeAac4rxAf8THSSQp5mRgG1ajTLNCgGn57wMBPzfwpFPRyAjV5kItlQ/4onamdZNNRBwV5YJT0eQmSlRCAb0jzEBVW84pmBEwEeasm8B5IARyIGQaa4iafHWuGiuIiJYYGuCwjLgYyIvplHmi5JJlGQSSiNIJgdEFiVOEUELJOgiZbp3pnm4iWDA6yTvYycEPXXmbJi0exrBGitzTvt4pqSQYIoAgg5KErHBDk4LgomAV8lYleRD+exlRO/Ftw8Y5k2x4mwonrwuJODzgNCsYzRXMVQ/D0zAI6hLrUyqBJ9BFvgstlE0e8AI0hWThfSTxc4Syrm9QgJ+EQY00pdVkVeETLfCRMC7A6LVVQh4bC+/Oxz3EVKC+7H7l2eREe47POFOJoxaDj4+iBPOvGgFJCFgTzISsBEaIp3xkJ/xzk68mAi4U8Uw8UP5OIkgfmQoicfDb5Am0sQvqQnjNSyxHhSbrxOwojdrMg+9gsQpHy2mnZGAr920sgStfnygFCfBVLqGsVeckOBtRGpchVqqBMzTzASjWXQqmvhaTVQIpr4vc9ChfTJRSyZsRTJOsdjxDrPvKa67Y2uM/H/s/zK/i4HPUvqzxEP6URKZmnvUkv4pSTJQXjajSRT3GnvizTv9AFjJ9eTRerlc2JxkGi1owirP3qzoA9emJeX/A0azrvJNMdx4AAAAAElFTkSuQmCC"></image-display>
```
Comment (chinook25) [inline apps/molgenis-components/src/components/display/ImageDisplay.vue:56]: maybe put the image in a separate file so this file is a bit more readable?

File: apps/styleguide/src/forms/InputRef.vue [hunk 12]
```diff
@@ -164,61 +166,61 @@ export default {
...
     deselect(key) {
       this.selection.splice(key, 1);
-      this.$emit('input', this.selection);
+      this.$emit("input", this.selection);
     },
     emitClear() {
-      if (this.list) this.$emit('input', []);
-      else this.$emit('input', null);
+      if (this.list) this.$emit("input", []);
+      else this.$emit("input", null);
```
Comment (chinook25) [inline apps/styleguide/src/forms/InputRef.vue:178]: missing curlies

File: apps/styleguide/src/forms/InputRef.vue [hunk 13]
```diff
@@ -164,61 +166,61 @@ export default {
...
       }
     },
     openSelect(idx) {
       this.showSelect = true;
       this.selectIdx = idx;
     },
     flattenObject(object) {
-      let result = '';
+      let result = "";
       Object.keys(object).forEach((key) => {
```
Comment (chinook25) [inline apps/styleguide/src/forms/InputRef.vue:200]: reduce

File: apps/styleguide/src/forms/InputRefSelect.vue [hunk 14]
```diff
@@ -112,14 +114,14 @@ export default {
...
           //nothing
-        } else if (typeof object[key] === 'object') {
+        } else if (typeof object[key] === "object") {
           result += this.flattenObject(object[key]);
         } else {
-          result += ' ' + object[key];
+          result += " " + object[key];
         }
       });
       return result;
```
Comment (chinook25) [inline apps/styleguide/src/forms/InputRefSelect.vue:127]: duplicate code from inputRef? Maybe extract?

File: apps/styleguide/src/mixins/TableMetadataMixin.vue [hunk 15]
```diff
@@ -60,29 +60,29 @@ export default {
...
+            (this.session.roles.includes("Editor") ||
+              this.session.roles.includes("Manager"))))
       );
     },
     canManage() {
       return (
         this.session &&
-        (this.session.email == 'admin' ||
-          this.session.roles.includes('Manager'))
+        (this.session.email == "admin" ||
```
Comment (chinook25) [inline apps/styleguide/src/mixins/TableMetadataMixin.vue:78]: ===

File: apps/styleguide/src/mixins/TableMixin.vue [hunk 16]
```diff
@@ -73,20 +75,20 @@ export default {
...
-            result = result + ' ' + col.id;
+            result = result + " " + col.id + "{" + this.refGraphql(col) + "}";
+          } else if (col.columnType == "FILE") {
+            result = result + " " + col.id + "{id,size,extension,url}";
+          } else if (col.columnType != "HEADING") {
+            result = result + " " + col.id;
           }
         });
       }
       return result;
```
Comment (chinook25) [inline apps/styleguide/src/mixins/TableMixin.vue:90]: reduce

File: apps/styleguide/src/mixins/TableMixin.vue [hunk 17]
```diff
@@ -114,12 +116,12 @@ export default {
...
+      let graphqlString = "";
       this.getTable(column.refTable).columns.forEach((c) => {
         if (c.key == 1) {
-          graphqlString += c.id + ' ';
+          graphqlString += c.id + " ";
           if (isReferentialColumn(c.columnType)) {
-            graphqlString += '{' + this.refGraphql(c) + '}';
+            graphqlString += "{" + this.refGraphql(c) + "}";
           }
         }
```
Comment (chinook25) [inline apps/styleguide/src/mixins/TableMixin.vue:126]: reduce

**Review by chinook25 — CHANGES_REQUESTED** [review]
Some minor things of which most will probably be 'not in this PR', but I thought I'd call them out just in case.

File: apps/.eslintrc.js (hunk 2 above)
Comment (mswertz) [reply apps/.eslintrc.js:36]: dito

File: apps/molgenis-components/lib/main.js [hunk 18]
```diff
@@ -47,12 +53,30 @@ import ObjectDisplay from "../src/components/tables/cellTypes/ObjectDisplay.vue"
...
 import Client from "../src/client/client.js";
 
+import EmailDisplay from "../src/components/tables/cellTypes/EmailDisplay.vue";
+import HyperlinkDisplay from "../src/components/tables/cellTypes/HyperlinkDisplay.vue";
+import InputHeading from "../src/components/forms/InputHeading.vue";
+import LayoutForm from "../src/components/layout/LayoutForm.vue";
+import MessageWarning from "../src/components/forms/MessageWarning.vue";
+import MolgenisMenu from "../src/components/layout/MolgenisMenu.vue";
+
+import LayoutCard from "../src/components/layout/LayoutCard.vue";
```
Comment (connoratrug) [inline apps/molgenis-components/lib/main.js:67]: please use folder order (see like the other components ) else its hard to keep track of inclusion.

File: apps/molgenis-components/package.json [hunk 19]
```diff
@@ -14,7 +15,8 @@
   },
   "scripts": {
     "dev": "vite",
-    "build": "cross-env SHOW_CASE=off vite build",
+    "ssr": "node server",
+    "build": "vite build",
```
Comment (connoratrug) [inline apps/molgenis-components/package.json:19]: why the change and addition

Comment (connoratrug) [inline apps/molgenis-components/package.json:19]: why the change and addition

File: apps/molgenis-components/lib/main.js (hunk 4 above)
Comment (mswertz) [reply apps/molgenis-components/lib/main.js:57]: I do agree but this currently we try to keep close to styleguide. Please leave unnecessary refactorings to later.

File: apps/build.gradle [hunk 20]
```diff
@@ -81,7 +81,7 @@ subprojects {
...
+    task buildComponentsShowcase(type: YarnTask, dependsOn: build) {
         node { with nodeSpec }
         if (project.name == 'molgenis-components') {
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/build.gradle:87]: this is a .gradle file written in groovy , the '==' means object equals there

File: apps/molgenis-components/src/main.js [hunk 21]
```diff
@@ -43,11 +45,12 @@ Object.entries(docs).forEach(([path, definition]) => {
     path: "/component/" + componentName,
     component: definition.default,
   }); // for detail view
-  const folderPath = path.split("/").slice(2); // remove folder root path
+  // docNames.push(componentName)
+  // for detail view
+  const folderPath = path.split("/").slice(4); // remove folder root path
   folderPath.pop(); // remove component name
```
[the remark this answers is not in the document]
Comment (connoratrug) [reply apps/molgenis-components/src/main.js]: not really , the const still points to the same array , the array is a mutable structure

File: apps/molgenis-components/src/components/display/ImageDisplay.vue (hunk 11 above)
Comment (connoratrug) [reply apps/molgenis-components/src/components/display/ImageDisplay.vue:56]: yes, but not a part of the current pr

File: apps/styleguide/src/forms/InputRef.vue (hunk 12 above)
Comment (connoratrug) [reply apps/styleguide/src/forms/InputRef.vue:178]: 👍

File: apps/central/src/components/admin/ManageSettings.vue [hunk 22]
```diff
@@ -82,70 +107,78 @@ export default {
...
-      this.isValueReadOnly = false,
-      this.actionFunction = this.createSetting
-      this.showModal = true
+      this.modalTitle = `Edit ${setting.key} setting`;
+      this.settingActionLabel = "Edit Setting";
+      this.settingKey = setting.key;
+      this.settingValue = setting.value;
+      (this.isKeyReadOnly = true),
+        (this.isValueReadOnly = false),
+        (this.actionFunction = this.createSetting);
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/central/src/components/admin/ManageSettings.vue:116]: this code was already like this @connoratrug ?

File: apps/molgenis-components/docs-plugin.js [hunk 23]
```diff
@@ -1,13 +1,42 @@
...
+const fs = require("fs");
 
-const transform = (code, id) => {
+module.exports = function () {
+  const transform = (code, id) => {
     if (/<docs>/.test(code)) {
-      console.log('remove docs from: ' + id);
-      // console.log(code.replace(/<docs>[\s\S.]*<\/docs>/g, ""));
+      // regex for custom docs tag
+      const re = /<docs>[\s\S.]*<\/docs>/;
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/docs-plugin.js]: renamed

File: apps/molgenis-components/docs-plugin.js [hunk 24]
```diff
@@ -1,13 +1,42 @@
...
+      const docs = code.match(re)[0];
+
+      // Build a path to store the docs
+      const pathSections = id.split("/");
+      const componentsPathIndex = pathSections.findIndex((section) => {
+        return section == "components";
+      });
+      const componentsPathSections = pathSections.splice(componentsPathIndex);
+      const componentFileName = componentsPathSections.pop();
+      const docPath = "./src/gen-docs/" + componentsPathSections.join("/");
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/docs-plugin.js]: why? I don't mind a bit of splitting of large functions but in this case I think it would make it harder to follow.

File: apps/molgenis-components/src/components/forms/IconAction.vue [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+  extends: ButtonAction,
+  props: {
+    label: String,
+    icon: String,
+  },
+};
+</script>
+
+<docs>
+
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/forms/IconAction.vue:25]: intellij does reasonable job though. Main question is if prettier can fix it?

File: apps/molgenis-components/src/main.js [hunk 26]
```diff
@@ -23,16 +24,17 @@ Object.entries(components).forEach(([path, definition]) => {
...
   { path: "/client", component: ClientView },
 ];
 
-const docsMap = {};
+let docsMap = {};
 
-// create routes for all the docs
-Object.entries(docs).forEach(([path, definition]) => {
+// create routes for generated docs
+Object.entries(genDocs).forEach(([path, definition]) => {
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/main.js:37]: wontfix for now

File: apps/molgenis-components/package.json (hunk 5 above)
Comment (mswertz) [reply apps/molgenis-components/package.json]: renamed back

File: apps/molgenis-components/src/components/tables/Pagination.vue [hunk 27]
```diff
@@ -107,20 +106,20 @@ export default {
...
-    data() {
-      return {
-        page: 3
-      }
-    }
-  }
-</script>
-
+<docs>
+  <template>
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/molgenis-components/src/components/tables/Pagination.vue:110]: has indeed to do with autoformatter don't doing its job properly :-(

File: apps/pages/src/components/ListPages.vue (hunk 8 above)
Comment (mswertz) [reply apps/pages/src/components/ListPages.vue:37]: some people say 'never use else'. I kind buy into this idea.

https://google.com/search?q=never%20use%20else

File: apps/molgenis-components/lib/main.js (hunk 18 above)
Comment (mswertz) [reply apps/molgenis-components/lib/main.js]: done

File: apps/molgenis-components/package.json (hunk 19 above)
Comment (mswertz) [reply apps/molgenis-components/package.json]: undone, has to do with ssr rendering of comp library that we might want do later

File: apps/central/src/components/admin/ManageSettings.vue (hunk 22 above)
[the remark this answers is not in the document]
Comment (chinook25) [reply apps/central/src/components/admin/ManageSettings.vue:116]: I think someone just made a typo, which accidentally is valid js and also does what we want :P

**Review by chinook25 — APPROVED** [review]
